test/e2e/e2e_suite_test.go

```diff
@@ -41,27 +42,31 @@ var _ = BeforeSuite(func() {
 
 	By("Creating a Cluster Proxy")
 	clusterProxy = framework.NewClusterProxy(initScheme())
+	kubeClient = clusterProxy.GetClient()
+	kubeconfigPath = clusterProxy.GetKubeconfigPath()
 
 	By("Creating a Keyvault Client")
 	kvClient = keyvault.NewClient(config)
 
 	if !config.IsSoakTest {
-		if !helm.ReleaseExists() { //If helm release exists, it means either cluster upgrade test or backward compatibility test is underway.
-			By(fmt.Sprintf("Installing Secrets Store CSI Driver and Azure Key Vault Provider via Helm from - %s.", config.HelmChartDir))
-			helm.Install(helm.InstallInput{
-				Config: config,
-			})
-		} else if config.IsBackwardCompatibilityTest {
-			//We upgrade only if we are running backward compatibility tests.
-			By(fmt.Sprintf("Upgrading Secrets Store CSI Driver and Azure Key Vault Provider via Helm to New Version from - %s.", config.HelmChartDir))
-			helm.Upgrade(helm.UpgradeInput{
-				Config: config,
-			})
+		if config.IsHelmTest {
```

Comment:
This is too many nested if else. I think we can return early on some of the scenarios and reduce the nesting.

1. If it's soak, return.
2. If it's not helm test, deploy with manifest and return.
3. If none of the above perform the checks for helm release exists.

Reply:
done